### ISSUE TITLE
Course instance table: Allow users to add notes to a Course

### ISSUE THREAD

**Issue by chrxr** [seas-computing/course-planner#453]
This ticket provides the ability for a user to add notes to a Course (as opposed to a Course Instance). 

This should utilize the Mark One modal and text area components.

Styling should be as demonstrated in the prototype: https://seas-computing.github.io/course-planner_ux/courses.html

The modal should include two buttons: "Cancel", and "Save"

## Cancel
* Clicking cancel, or clicking outside of the modal will trigger the unsaved changes notifications.
* Confirming unsaved changes will close the modal without any changes to the data.

## Save
* Clicking "Save" will send the value of the text area field to the `Course` PUT request endpoint.
* This ticket will require updating the `Course` PUT request endpoint to include the `notes` field.

### Validation

* On submit of the notes data, we should trim blank spaces.

**Comment by rmainseas** [comment]
I do just want to write up some notes here about where things are and where I've got to so far...

Due to the way the table is built (for each row => for each cell), I ended up having to pass the `openNotesModal` function through the loop like that in order to keep the modal component in the `CoursesPage` component and pass the toggle function through into the cell correctly. I spent a while trying to figure out if I could bypass the table building loop and pass it in directly, but I couldn't find a way to get that to work.

The code to open the modal is currently working and I can display notes and information from the `CourseInstance`(i would've liked to pass a `Course` object in rather than ` CourseInstance` for semantics but the `CourseInstance`object seemed to have the data I was after for now, so I ended up using that instead).
